core/src/main/java/org/elasticsearch/index/translog/TranslogDeletionPolicy.java

```diff
@@ -154,4 +153,8 @@ private long getMinTranslogGenRequiredByViews() {
     public synchronized long getMinTranslogGenerationForRecovery() {
         return minTranslogGenerationForRecovery;
     }
+
+    synchronized long getViewCount(long viewGen) {
+        return translogRefCounts.getOrDefault(viewGen, Counter.newCounter(false)).get();
```

Comment:
you are creating a new counter anyway even if it's there I think we can safe that object and use `computeIfAbsent`?